inotify.go

```diff
@@ -74,6 +104,11 @@ func (w *Watcher) Close() error {
 	if w.isClosed() {
 		return nil
 	}
+	limits.Lock()
+	if limits.max != 0 {
```

Comment:
I just noticed a bug here. We should always decrement limits.allocated, not just when limits.max != 0 because we always increment limits.allocated. If someone calls SetMaxWatcherLimit(100) after some watchers have been created and dropped, the limits.allocated may be permanently above 0.

Reply:
whoops, you are indeed correct